packages/flutter/lib/src/rendering/editable_line.dart

```diff
@@ -111,6 +116,18 @@ class RenderEditableLine extends RenderBox {
     markNeedsPaint();
   }
 
+  List<Point> getEndpointsForSelection(TextSelection selection) {
+    List<ui.TextBox> selectionRects = _textPainter.getBoxesForSelection(selection);
+    Offset offset = _paintOffset + new Offset(0.0, -_kCaretHeightOffset);
+    Point left = selectionRects.first.toRect().bottomLeft + offset;
+    Point right = selectionRects.last.toRect().bottomRight + offset;
+    return [localToGlobal(left), localToGlobal(right)];
```

Comment:
This isn't quite right for TRL.  You need to look at the textDirection of each rect to decide whether to take the left or right side.  For the first point, you want the leading side of the first rect (i.e., left for LTR and right for RTL).  For the second point, you want the trailing side of the last rect (i.e., right for LTR and left for RTL).

Also, just returning two points isn't always sufficient.  Some visual styles need to know the text direction at those points.  For example, if the selection handles aren't symmetric, they might want to orient them depending on the text direction at that point.